### PR TITLE
Add NetBSD support

### PR DESCRIPTION
**Pull request by coypoop**
Mostly picking the same paths as FreeBSD.
We need a little special handling for crt files, as netbsd uses its
own (and not GCC's) for those, with slightly different names.

While running the tests I think it's building the self-hosted zig, I had to explicitly link against /usr/lib/libstdc++.a:

env LD_LIBRARY_PATH=/usr/pkg/lib /home/fly/zig/build/bin/zig build-exe /home/fly/zig/src-self-hosted/main.zig --object /home/fly/zig/build/zig_cpp/libzig_cpp.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_wasm.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_elf.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_coff.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_lib.a --cache-dir /home/fly/zig/zig-cache --output /home/fly/zig/zig-cache/zig --name zig --library c --library LLVM-7  -isystem /home/fly/zig/src -isystem /home/fly/zig/build --object /usr/lib/libstdc++.a  -isystem /usr/pkg/include --library-path /usr/pkg/lib

Otherwise it was emitting errors like:
````
lld: error: undefined symbol: operator delete(void*)
>>> referenced by OutputSections.h:102 (/home/fly/zig/deps/lld/wasm/OutputSections.h:102)
>>>               OutputSections.cpp.o:(lld::wasm::DataSection::~DataSection()) in archive /home/fly/zig/build/zig_cpp/libembedded_lld_wasm.a

lld: error: undefined symbol: std::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()
>>> referenced by OutputSections.h:88 (/home/fly/zig/deps/lld/wasm/OutputSections.h:88)
>>>               OutputSections.cpp.o:(lld::wasm::CodeSection::~CodeSection()) in archive /home/fly/zig/build/zig_cpp/libembedded_lld_wasm.a

lld: error: undefined symbol: operator delete(void*)
>>> referenced by OutputSections.h:88 (/home/fly/zig/deps/lld/wasm/OutputSections.h:88)
>>>               OutputSections.cpp.o:(lld::wasm::CodeSection::~CodeSection()) in archive /home/fly/zig/build/zig_cpp/libembedded_lld_wasm.a

The following command exited with error code 1:
/home/fly/zig/build/bin/zig build-exe /home/fly/zig/src-self-hosted/main.zig --object /home/fly/zig/build/zig_cpp/libzig_cpp.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_wasm.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_elf.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_coff.a --object /home/fly/zig/build/zig_cpp/libembedded_lld_lib.a --cache-dir /home/fly/zig/zig-cache --output /home/fly/zig/zig-cache/zig --name zig --library c --library LLVM-7 -isystem /home/fly/zig/src -isystem /home/fly/zig/build -isystem /usr/pkg/include --library-path /usr/pkg/lib

Build failed. The following command failed:
/home/fly/.local/share/zig/stage1/artifact/UYlxOUzNCbMTc5_2CnyaMaY4eSw8e8wMEwO0FiSkJ1Qe03FlWCmKZJ8MFNdrhJGV/build /home/fly/zig/build/bin/zig /home/fly/zig /home/fly/zig/zig-cache test
````

(Also, the LD_LIBRARY_PATH is a substitute for an RPATH for libLLVM-7.so)

### CODE REVIEW

**Comment by coypoop** [comment]
I haven't found out whether it's safe to do https://github.com/ziglang/zig/pull/1972/commits/bc10382ec1b87da16943907ba2d0fbd267af07f0#diff-3ce0315ddf9ec17d0e8a5f5a4fcaf8b7R286 but it hasn't bitten me yet

**Comment by andrewrk** [comment]
Thanks for sending the patch upstream! I enjoyed reading your blog article.

I'll have time to go over this pull request tomorrow (Monday).

Can you confirm that you double checked all the structs and function prototypes with your system header files (as opposed to copying them straight from FreeBSD)? It's OK if you didn't - we can either delete the not-double-checked ones if they're unused, or I don't mind doing it, I'll just have to rent a VPN for an hour. I just want to avoid something like 7a4ed10981cf7ac5bed93b2c9bad280ad31c50cf which was not a straightforward debug process.

**Comment by andrewrk** [comment]
I believe `EVFILT_USER` is necessary for `std.event.Loop` to function correctly, but we're all waiting for copy-elision/coroutine rewrite before getting back into event-based code. I do think it's better to not define a meaningless constant and set it to 0 - that's surprising behavior. Better behavior is to leave the places that try to use EVFILT_USER a compile error, and then disable the tests for that code. Don't pretend it works if it doesn't!

**Comment by coypoop** [comment]
Yes, I reviewed all the structs and defines rather than copying.

**Comment by andrewrk** [comment]
Great work! Thanks for your contribution. 

Looks like we need to go add a new column to the README support table, thanks to you.

File: std/os/netbsd/errno.zig [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+pub const ESRCH = 3; // No such process
+pub const EINTR = 4; // Interrupted system call
+pub const EIO = 5; // Input/output error
+pub const ENXIO = 6; // Device not configured
+pub const E2BIG = 7; // Argument list too long
+pub const ENOEXEC = 8; // Exec format error
+pub const EBADF = 9; // Bad file descriptor
+pub const ECHILD = 10; // No child processes
+pub const EDEADLK = 11; // Resource deadlock avoided
+// 11 was EAGAIN
```
Comment (daurnimator) [inline std/os/netbsd/errno.zig]: What do you mean "was"? It seems to be 35 now.
When did it change? Why did it change?

File: std/os/netbsd/errno.zig [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+pub const ENOTCONN = 57; // Socket is not connected
+pub const ESHUTDOWN = 58; // Can't send after socket shutdown
+pub const ETOOMANYREFS = 59; // Too many references: can't splice
+pub const ETIMEDOUT = 60; // Operation timed out
+pub const ECONNREFUSED = 61; // Connection refused
+
+pub const ELOOP = 62; // Too many levels of symbolic links
+pub const ENAMETOOLONG = 63; // File name too long
+
+// should be rearranged
```
Comment (daurnimator) [inline std/os/netbsd/errno.zig]: Meaning of this comment?

Comment (coypoop) [reply std/os/netbsd/errno.zig]: Sorry, I pulled it straight out of the [errno.h](https://nxr.netbsd.org/xref/src/sys/sys/errno.h#117) header in netbsd. I assumed the comments are going to be meaningful, but missed that this one isn't

File: std/os/netbsd/errno.zig (hunk 1 above)
Comment (Aversiste) [reply std/os/netbsd/errno.zig]: It is a comment from the original file in /usr/include/sys/errno.h, same thing on OpenBSD.

**Pull request merged**